app/controllers/publishers/omniauth_callbacks_controller.rb

```diff
@@ -17,7 +17,11 @@ def dfe
   end
 
   def failure
-    redirect_to root_path
+    omniauth_error = request.respond_to?(:get_header) ? request.get_header("omniauth.error") : request.env["omniauth.error"]
+    Rollbar.error(omniauth_error, strategy: failed_strategy.name)
+    Rails.logger.error("DSI failure - strategy: #{failed_strategy.name}, reason: #{omniauth_error.inspect}")
+
+    redirect_to new_publisher_session_path, flash: { warning: t(".message") }
```

Comment:
When you redirect like this does it mean you can get a warning flash without having it in `add_flash_types`?

Reply:
It's in `add_flash_types` now that Alex has added it (as of a few minutes ago).

The problem is that `redirect_to` will only accept either `alert` or `notice`, not any of the custom types we have defined, so we need to nest it under the `flash` key. 😿